release/Magarena/scripts/Din_of_the_Fireherd.groovy

```diff
@@ -0,0 +1,40 @@
+def action = {
+    final MagicGame game, final MagicEvent event ->
+            event.processTargetPlayer(game, {
+                final MagicPlayer player = event.getPlayer();
+                final int BC = BLACK_CREATURE_YOU_CONTROL.filter(player).size();
+                game.logAppendValue(player,BC)
+                for (int i = 0; i < BC; i++) {
+                    game.addEvent(new MagicSacrificePermanentEvent(event.getSource(), it, SACRIFICE_CREATURE));
+                }
+                final int RC = RED_CREATURE_YOU_CONTROL.filter(player).size();
+                game.logAppendValue(player,RC)
+                for (int i = 0; i < RC; i++) {
+                    game.addEvent(new MagicSacrificePermanentEvent(event.getSource(), it, SACRIFICE_LAND));
+        }
+    });
+}
+
+[
+    new MagicSpellCardEvent() {
+        @Override
+        public MagicEvent getEvent(final MagicCardOnStack cardOnStack,final MagicPayedCost payedCost) {
+            return new MagicEvent(
```

Comment:
target opponent should be chosen in this event as this is a targeted spell

Reply:
You mean moving the target selection from the subevent to the main spell?
True, opponent could have hexproof for example, so this might make a difference.
And you should not be able to cast that spell in that case (no valid targets)

Reply:
Yes, the target selection should be in the main event of the spell.